Lib/pdb.py

```diff
@@ -188,9 +189,27 @@ def __init__(self, completekey='tab', stdin=None, stdout=None, skip=None,
     def sigint_handler(self, signum, frame):
         if self.allow_kbdint:
             raise KeyboardInterrupt
-        self.message("\nProgram interrupted. (Use 'cont' to resume).")
+        try:
+            self.message("\nProgram interrupted (please wait).")
+        except RuntimeError:
+            # 'print' is not safe in signal handlers, check issue24283
+            # if we encounter this issue, we just silently resume the execution
+            pass
         self.set_step()
-        self.set_trace(frame)
+        if self.sigint_triggered_cmd == 'continue':
+            self.set_trace(frame)
```

Comment:
Why is this only needed / special for "continue"?  Because it might have unset the tracing function (when there are no breakpoints)?

btw: `Pdb/Bdb.set_trace` calls `set_step` itself, so it might not be needed here explicitly before then.

Reply:
Because `set_continue()` does not stop the debugger in an `interaction()` on the next debug event.

Thanks for your suggestion, it works.